main.go

```diff
@@ -86,30 +94,128 @@ func isDockerCommand(kw string) bool {
 	return false
 }
 
-func completer(d prompt.Document) []prompt.Suggest {
-	word := d.GetWordBeforeCursor()
+//DockerHubResult : Wrap DockerHub API call
+type DockerHubResult struct {
+	PageCount        *int             `json:"num_pages,omitempty"`
+	ResultCount      *int             `json:"num_results,omitempty"`
+	ItemCountPerPage *int             `json:"page_size,omitempty"`
+	CurrentPage      *int             `json:"page,omitempty"`
+	Query            *string          `json:"query,omitempty"`
+	Items            []DockerHubImage `json:"results,omitempty"`
+}
+
+//DockerHubImage : Wrap image results of DockerHub API call
+type DockerHubImage struct {
+	Name        string `json:"name,omitempty"`
+	Description string `json:"description,omitempty"`
+	StarCount   int    `json:"star_count,omitempty"`
+	IsTrusted   *bool  `json:"is_trusted,omitempty"`
+	IsAutomated bool   `json:"is_automated,omitempty"`
+	IsOffical   *bool  `json:"is_official,omitempty"`
+}
+
+func imageFetchCompleter(imageName string, count int) []prompt.Suggest {
+	url := url.URL{
+		Scheme:   "https",
+		Host:     "registry.hub.docker.com",
+		Path:     "/v1/search",
+		RawQuery: "q=" + url.PathEscape(imageName) + "&n=" + strconv.Itoa(count),
+	}
+	if imageName == "" {
+		url.Path = "/v2/repositories/library"
+		url.RawQuery = "page=1&page_size=" + strconv.Itoa(count)
+	}
+	client := &http.Client{}
+	apiURL := url.String()
 
-	for _, cmd := range strings.Split(d.Text, " ") {
-		if strings.HasPrefix(cmd, "-") || strings.HasPrefix(cmd, "/") || cmd == "" {
+	req, err := http.NewRequest("GET", apiURL, nil)
+	if err != nil {
+		return nil
+	}
+	req.Header.Add("Content-Type", "application/json")
+	response, err := client.Do(req)
+	if err != nil {
+		return nil
+	}
+
+	defer response.Body.Close()
+
+	decoder := json.NewDecoder(response.Body)
+	searchResult := &DockerHubResult{}
+	decoder.Decode(searchResult)
+
+	suggestions := []prompt.Suggest{}
+	for _, s := range searchResult.Items {
+		if s.IsOffical != nil {
+			description := "Not Offical"
+			if *s.IsOffical {
+				description = "Offical"
+			}
+			suggestions = append(suggestions, prompt.Suggest{Text: s.Name, Description: "(" + description + ") " + s.Description})
 			continue
 		}
+		suggestions = append(suggestions, prompt.Suggest{Text: s.Name, Description: s.Description})
+	}
```

Comment:
When we replace IsOffical pointer with non pointer type, we can simply refactor here as:
```suggestion
	for _, s := range searchResult.Items {
		suggestions = append(suggestions, prompt.Suggest{Text: s.Name, Description: "(" + dockerDescription[s.IsOffical] + ") " + s.Description})
	}

dockerDescription is a map which holds mapping true: Offical false: Not Offical (or we can keep inner if condition and remove s.IsOffical != nil logic)
```

Reply:
If the official status is not important, the pointer would be removed and the refactoring would be applied.